### PR TITLE
solution hw-spring-06

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

**Review by MaliukDaria — APPROVED** [review]

File: src/main/java/com/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.stream.Collectors;
+import org.modelmapper.ModelMapper;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("cinema-halls")
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controller/CinemaHallController.java]:
```suggestion
@RequestMapping("/cinema-halls")
```
let's add this everywhere

File: src/main/java/com/cinema/mapper/OrderMapper.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public Order fromDtoToEntity(OrderRequestDto requestDto) {
+        User user = userService.get(requestDto.getUserId());
+        ShoppingCart userCart = shoppingCartService.getByUser(user);
+        Order order = new Order();
+        order.setOrderDate(LocalDateTime.now());
+        order.setTickets(new ArrayList<>(userCart.getTickets()));
+        order.setUser(user);
+        return order;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/mapper/OrderMapper.java]: if we move this logic here, we will violate the single responsibility principle

take it back to service layer

File: src/main/java/com/cinema/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private final UserService userService;
+
+    public OrderController(OrderService orderService, OrderMapper mapper, UserService service) {
+        this.orderService = orderService;
+        this.mapper = mapper;
+        this.userService = service;
+    }
+
+    @PostMapping("/complete")
+    public void create(@RequestBody OrderRequestDto orderRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controller/OrderController.java]: let's pass userId as a @RequestParam?

File: src/main/java/com/cinema/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                                  UserService userService, ShoppingCartMapper mapper) {
+        this.cartService = cartService;
+        this.sessionService = sessionService;
+        this.userService = userService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestParam(name = "id") Long movieSessionId,
+                                @RequestParam String email) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controller/ShoppingCartController.java]:
```suggestion
                                @RequestParam Long userId) {
```

File: src/main/java/com/cinema/dao/MovieSessionDao.java [hunk 5]
```diff
@@ -8,4 +8,6 @@
     List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
 
     MovieSession add(MovieSession session);
+
+    MovieSession findById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/MovieSessionDao.java]:
```suggestion
    MovieSession get(Long id);
```
uniform method names

File: src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package com.cinema.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long shoppingCartId;
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java]: aren't these the same?

File: src/main/java/com/cinema/model/dto/order/OrderResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
+package com.cinema.model.dto.order;
+
+import com.cinema.model.dto.TicketResponseDto;
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private LocalDateTime localDateTime;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/order/OrderResponseDto.java]:
```suggestion
    private LocalDateTime orderCreationTime;
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/controller/CinemaHallController.java (hunk 1 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/controller/CinemaHallController.java]: done

File: src/main/java/com/cinema/mapper/OrderMapper.java (hunk 2 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/mapper/OrderMapper.java]: done

File: src/main/java/com/cinema/controller/ShoppingCartController.java (hunk 4 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/controller/ShoppingCartController.java]: done

File: src/main/java/com/cinema/model/dto/order/OrderResponseDto.java (hunk 7 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/model/dto/order/OrderResponseDto.java]: done

File: src/main/java/com/cinema/model/dto/order/OrderRequestDto.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.cinema.model.dto.order;
+
+public class OrderRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/order/OrderRequestDto.java]: do we use this now?

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/cinema/model/dto/order/OrderRequestDto.java (hunk 8 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/model/dto/order/OrderRequestDto.java]: no. deleted

**Pull request merged**